src/models/generators/musicXmlFile.coffee

```diff
@@ -0,0 +1,130 @@
+promisify = require("bluebird").promisifyAll
+fs = promisify require("fs")
+o2x = require('object-to-xml');
+_ = require("protolodash")
+
+###
+A MusicXml File created from a *score*.
+###
+module.exports =
+
+class MusicXmlFile
+  constructor: (score) ->
+    @file =
+      '?xml version=\"1.0\" encoding=\"UTF-8\"?' : null,
+      'score-partwise' :
+        '#' :
+          'part-list' :
+            'score-part' :
+              '@' :
+                id : 'P1'
+              ,
+              '#' :
+                'part-name': 'MusicXML Part'
+
+          part :
+            '@' :
+              id : 'P1'
+            ,
+            '#' :
+              @_mapBars(score)
+
+  ###
+  Maps the bars into XML notation.
+  ###
+  _mapBars: (score) =>
+
+    measureNum = 0
+    measures = {}
+
+    measures = for bar in score.bars
+      measureNum += 1
+      {
+        '@' :
+          'number' : measureNum
+        '#' :
+          'attributes' : if (measureNum == 1)
+            @_getAtributes(bar)
+          'note' : [@_mapNotes(bar.notes)]
+      }
+
+    mappedBars = 'measure' : [
+      measures
+    ]
+
+
+  ###
+  Get atributes.
+  ###
+  _getAtributes: (bar) =>
+    {
+      'divisions' : 1,
+      'key':
+        'fifths': 0,
+        'mode': 'major'
+      'time':
+        'beats' : bar.signatures.time.major
+        'beat-type' : bar.signatures.time.minor
+      'clef':
+        'sign' : bar.signatures.clef
+        'line' : 2
+    }
+
+  ###
+  Maps the notes.
+  ###
+  _mapNotes: (notes) =>
+    for note in notes
```

Comment:
Idem. Acá usaría el `map` en vez de mapear las notas con un for (que solo funciona en coffee).

Con algunos shortcuts esto se puede escribir:
```coffee-script
    notes.map (note) =>
      if note.name != "r"
        mappedNote =
          pitch:
            step: note.name.charAt(0).toUpperCase()
            octave: note.name.slice(-1)
          duration: 1
          voice: 1
          type: @_noteType note.duration

        if note.name.charAt(1) is "#"
          _.assign mappedNote.pitch, { alter: "1" }
        if note.name.charAt(1) is "b"
          _.assign mappedNote.pitch, { alter: "-1" }

        mappedNote
      else
        rest: null
        duration: 1
        voice: 1
        type: @_noteType note.duration
```

Fijate que:
- no hace falta asignar una variable pa' retornarla
- las llaves son opcionales, y los paréntesis en muchos casos
- se puede usar `is`, `isnt`, y `not` en lugar de `==`, `!=` y `!`